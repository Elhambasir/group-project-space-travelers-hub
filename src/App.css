.m-2{
  text-decoration: none;
}

.link{
  text-decoration: none;
}

.active-link {
  text-decoration: underline;
}

.Logo {
  width: 50px;
}

.Line {
  border: solid rgba(128, 128, 128, 0.395) 1px;
  width: auto;
  margin-left: 30px;
  margin-right: 30px;
}
