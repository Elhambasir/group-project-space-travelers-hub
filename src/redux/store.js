// eslint-disable-next-line import/no-extraneous-dependencies
import { configureStore } from '@reduxjs/toolkit';
import MissionReducer from './mission/missionSlice';

// eslint-disable-next-line import/prefer-default-export
export const store = configureStore({
  reducer: {
    missions: MissionReducer,
  },
});
