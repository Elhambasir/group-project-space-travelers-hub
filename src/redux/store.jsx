// eslint-disable-next-line import/no-extraneous-dependencies
import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './Rockets/rocketsSlice';
import MissionReducer from './mission/missionSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: MissionReducer,
  },
});

export default store;
