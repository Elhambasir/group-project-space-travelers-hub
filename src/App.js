import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Nav from './components/Nav';
import './App.css';

function App() {
  return (
    <>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Nav />} />
          <Route path="*" element={<h3>No Match</h3>} />
        </Routes>
      </BrowserRouter>
    </>
  );
}

export default App;
