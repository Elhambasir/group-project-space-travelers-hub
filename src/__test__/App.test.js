import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/store';

describe('Component render testing', () => {
  it('Line page renders correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
