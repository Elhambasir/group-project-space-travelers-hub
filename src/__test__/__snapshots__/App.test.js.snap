// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render testing Line page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-wrap align-items-center p-3 container"
  >
    <img
      alt="Logo"
      class="Logo"
      src="https://spacetravellerhubcapstone.netlify.app/static/media/planet.3f6fb2742f28651730c0.png"
    />
    <h3>
      Space Travelers' Hub
    </h3>
    <ul
      class="list-unstyled m-2 p-2 d-flex flex-wrap"
      style="justify-content: flex-end; flex: 2;"
    >
      <li
        class="m-2"
      >
        <a
          class="link"
          href="/Rockets"
          style="text-decoration: none;"
        >
          Rockets
        </a>
      </li>
      <li
        class="m-2"
      >
        <a
          class="link"
          href="/Mission"
          style="text-decoration: none;"
        >
          Mission
        </a>
      </li>
      <li
        class="m-2"
      >
        <a
          class="link"
          href="/MyProfile"
          style="text-decoration: none;"
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
  <div
    class="Line"
  />
  <h3>
    No Match
  </h3>
</DocumentFragment>
`;
